lib/improver/blending/weights.py

```diff
@@ -493,13 +493,30 @@ def _create_new_weights_cube(self, cube, weights):
         for cube_slice, weight in (
                 zip(cube.slices_over(self.weighting_coord_name), weights)):
             sub_slice = cube_slice[..., 0, 0]
-            sub_slice.remove_coord(sub_slice.coord(axis='x'))
-            sub_slice.remove_coord(sub_slice.coord(axis='y'))
             sub_slice.data = np.ones(sub_slice.data.shape)*weight
             cubelist.append(sub_slice)
+
+        # re-order dimension coordinates to match input cube
         new_weights_cube = (
             check_cube_coordinates(cube[..., 0, 0], cubelist.merge_cube()))
+
+        # remove all scalar coordinates that are not time-, model- or
+        # blend-related
+        dim_coords = new_weights_cube.coords(dim_coords=True)
```

Comment:
Do any of the unit tests capture this change? If not should we write one?

Reply:
None of the unit tests capture this change, no.  Unfortunately, I can't really just write one that will capture the change, because the cubes that are being set up by the centralised function are not representative of what would actually ever be passed in to the plugin.  I could add in a cludgy individual test that sets up sensible cubes and then tests for removal of those coordinates.  Or I could rewrite the lot (which would be better, but might obscure the impact of this change and would be more work to review).

Reply:
No, the unit tests don't really cover anything useful given the input cube setup.  I could write a cludgy test that does its own setup and cover this, or I could refactor the module to deal with sensible inputs.  Given this example of the tests being un-extendable, it's probably worth me having a go at the latter.